doc/ideas/savefileformat.md

```diff
@@ -0,0 +1,446 @@
+# openage save format
+
+1.  [Questions](#1-questions)
+2.  [Overview](#2-overview)
+3.  [Header](#3-header)
+4.  [String Table](#4-string-table)
+5.  [Atrribute Table](#5-attribute-table)
+6.  [Game Information](#6-game-information)
+7.  [User Information](#7-user-information)
+8.  [Chunk Table](#8-chunk-table)
+9.  [Single Chunks](#9-single-chunks)
+10. [Triggers](#10-triggers)
+    1. [Event List](#101-event-list)
+    2. [Condition List](#102-condition-list)
+    3. [Action List](#103-action-list)
+11. [Crypto](#11-crypto)
+
+### 1 Questions
+
+Do we something like ingame time vs realtime if the game is played at a different speed?
+
+What do we want to track for stats?
+
+| category   | stats  |
+| ---------- | ------ |
+| ressources | collected food/wood/gold/stone, maximum res in the bank? |
+| units      | units/buildings killed/lost, units converted/lost to convertions, max units of each type |
+| map        | discovered, maximum simultanously viewable (discovered - fog of war) |
+| tech       | when was each tech/age researched? |
+| battle     | what is a battle? units killed per timeslot per close area 30 units died in 15 seconds in radius of 20 Tiles TODO: definitions, maybe add cost for unit for calculation of the size of battle |
+
+do we record timeseries? if yes fixed sizes(round robin database) or simple array snapshot per 10s/30s/60s/flexible?
+
+every minute we save unit count per player?
+
+crypto? multiplayer safe => server creates savefile, encrypts it, but one member of every team is needed to decrypt it and continue the game https://en.wikipedia.org/wiki/Secret_sharing
+
+- savefile vs recorded game what happens if a savefile is continued, new recorded game or should recorded game stored with the savefile? if we dont store it, maybe we can reference it, so it can be continued if available
```

Comment:
I think it would be possible to store the whole history in a file, just whenever a keyframe is surpassed by the renderer, it can be written into the savefile.
If we save the game, all "future" keyframes are written, the queues are written and the current time is stored.